carma_cooperative_perception/src/external_object_list_to_detection_list_component.cpp

```diff
@@ -0,0 +1,241 @@
+
+      if (current_state == "active") {
+        update_georeference(*msg_ptr);
+      } else {
+        RCLCPP_WARN(
+          this->get_logger(),
+          "Trying to receive message on the topic '%s', but the containing node is not activated. "
+          "Current node state: '%s'",
+          this->georeference_subscription_->get_topic_name(), current_state.c_str());
+      }
+    });
+
+  declare_parameter(
+    "small_vehicle_motion_model",
+    carma_cooperative_perception_interfaces::msg::Detection::MOTION_MODEL_CTRV);
+
+  declare_parameter(
+    "large_vehicle_motion_model",
+    carma_cooperative_perception_interfaces::msg::Detection::MOTION_MODEL_CTRV);
+
+  declare_parameter(
+    "motorcycle_motion_model",
+    carma_cooperative_perception_interfaces::msg::Detection::MOTION_MODEL_CTRA);
+
+  declare_parameter(
+    "pedestrian_motion_model",
+    carma_cooperative_perception_interfaces::msg::Detection::MOTION_MODEL_CV);
+
+  declare_parameter(
+    "unknown_motion_model",
+    carma_cooperative_perception_interfaces::msg::Detection::MOTION_MODEL_CV);
+
+  on_set_parameters_callback_ =
+    add_on_set_parameters_callback([this](const std::vector<rclcpp::Parameter> & parameters) {
+      rcl_interfaces::msg::SetParametersResult result;
+      result.successful = true;
+      result.reason = "success";
+
+      for (const auto & parameter : parameters) {
+        if (parameter.get_name() == "small_vehicle_motion_model") {
+          this->motion_model_mapping_.small_vehicle_model =
+            static_cast<std::uint8_t>(parameter.as_int());
+        } else if (parameter.get_name() == "large_vehicle_motion_model") {
+          this->motion_model_mapping_.large_vehicle_model =
+            static_cast<std::uint8_t>(parameter.as_int());
+        } else if (parameter.get_name() == "motorcycle_motion_model") {
+          this->motion_model_mapping_.motorcycle_model =
+            static_cast<std::uint8_t>(parameter.as_int());
+        } else if (parameter.get_name() == "pedestrian_motion_model") {
+          this->motion_model_mapping_.pedestrian_model =
+            static_cast<std::uint8_t>(parameter.as_int());
+        } else if (parameter.get_name() == "unknown_motion_model") {
+          this->motion_model_mapping_.unknown_model = static_cast<std::uint8_t>(parameter.as_int());
+        } else {
+          result.successful = false;
+          result.reason = "Unexpected parameter name '" + parameter.get_name() + '\'';
+        }
+      }
+
+      return result;
+    });
+
+  return carma_ros2_utils::CallbackReturn::SUCCESS;
+}
+
+auto ExternalObjectListToDetectionListNode::handle_on_activate(
+  const rclcpp_lifecycle::State & /* previous_state */) -> carma_ros2_utils::CallbackReturn
+{
+  RCLCPP_INFO(get_logger(), "Life cycle state transition: activating");
+
+  publisher_->on_activate();
+
+  return carma_ros2_utils::CallbackReturn::SUCCESS;
+}
+
+auto ExternalObjectListToDetectionListNode::handle_on_deactivate(
```

Comment:
It looks like implementations of handle_on_deactivate, handle_on_cleanup, and handle_on_shutdown 
 in `CarmaLifecycleNode` might already handle the work that is being completed in the functions here. I might be missing something, but could you please confirm and/or describe why the function overrides are needed for these?